### PR TITLE
Port change from https://github.com/dotnet/core-setup/pull/8426

### LINKED ISSUE #533: Port change from Core-Setup to rely on Arcade for dotnetcli blob storage publishing
See https://github.com/dotnet/core-sdk/pull/6473#issuecomment-589389674

This repo needs to port https://github.com/dotnet/core-setup/pull/8426 to be more flexible with new channels being added.

### PR DESCRIPTION
**Pull request by AdamYoblick**
Fixes #533

### CODE REVIEW

File: publish/prepare-artifacts.proj [hunk 1]
```diff
@@ -1,6 +1,24 @@
 <Project>
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
+	<UsingTask TaskName="GenerateChecksums" AssemblyFile="$(LocalBuildToolsTaskFile)" />
```
Comment (dagood) [inline publish/prepare-artifacts.proj:4]: A few tabs snuck in.

File: publish/prepare-artifacts.proj [hunk 2]
```diff
@@ -13,12 +31,17 @@
...
             FindDownloadedArtifacts;
-            SignPackages">
+            SignPackages;
+            CreateChecksums">
     <PropertyGroup>
       <PreparedFileUploadDir>$(ArtifactsObjDir)PreparedFileUpload\</PreparedFileUploadDir>
     </PropertyGroup>
 
-    <Copy SourceFiles="@(UploadToBlobStorageFile)" DestinationFolder="$(PreparedFileUploadDir)">
+	  <ItemGroup>
```
Comment (dagood) [inline publish/prepare-artifacts.proj:40]: A few tabs snuck in.

File: publish/prepare-artifacts.proj [hunk 3]
```diff
@@ -96,10 +119,13 @@
...
-      <ItemsToPush>
-        <RelativeBlobPath>assets/windowsdesktop/$(InstallersRelativePath)%(Filename)%(Extension)</RelativeBlobPath>
+        Exclude="@(NupkgToPublishFile);@(SymbolNupkgToPublishFile)">
+        <RelativeBlobPath>$(InstallersRelativePath)%(Filename)%(Extension)</RelativeBlobPath>
       </ItemsToPush>
+
+      <ItemsToPush Include="@(GeneratedChecksumFile)">
+        <RelativeBlobPath>$(InstallersRelativePath)%(Filename)%(Extension)</RelativeBlobPath>
+        <Category>Checksum</Category>
     </ItemGroup>
```
Comment (dagood) [inline publish/prepare-artifacts.proj:129]: Syntax error, unclosed element.

File: publish/prepare-artifacts.proj (hunk 1 above)
Comment (AdamYoblick) [reply publish/prepare-artifacts.proj:4]: Ugh, copy paste from web browser will do that 😆

File: publish/prepare-artifacts.proj (hunk 2 above)
Comment (AdamYoblick) [reply publish/prepare-artifacts.proj]: Fixed

File: publish/prepare-artifacts.proj (hunk 3 above)
Comment (AdamYoblick) [reply publish/prepare-artifacts.proj]: Fixed

**Comment by AdamYoblick** [comment]
How do we do a mock official build? I'm not familiar with this process

**Comment by mmitche** [comment]
@AdamYoblick push your branch to the internal dnceng instance and associate it using darc add-default-channel to the "General Testing" channel. Be sure to use the AzDO repo uri:

https://dev.azure.com/dnceng/internal/_git/dotnet-windowsdesktop

Then run the official build.

**Comment by AdamYoblick** [comment]
will do, thank you Matt

**Comment by dagood** [comment]
Missing `publish/publish-blobs.proj` deletion?
https://github.com/dotnet/core-setup/pull/8426/files#diff-683bea6393d11dd85e95d8ff427add64

**Comment by AdamYoblick** [comment]
Yep, sorry. Trying to multitask and messing everything up. Will fix now.

**Comment by AdamYoblick** [comment]
* code-mirror pipeline is running at https://dnceng.visualstudio.com/internal/_build/results?buildId=642047&view=results.
* default channel has been added:
  * (1602) https://dev.azure.com/dnceng/internal/_git/dotnet-windowsdesktop @ port_core_setup_change -> General Testing

Once the mirror is done, I'll kick off the build

**Comment by AdamYoblick** [comment]
Official windowsdesktop build for the port_core_setup_change branch is running at https://dnceng.visualstudio.com/internal/_build/results?buildId=642072&view=results

**Comment by AdamYoblick** [comment]
@dagood @mmitche Looks like the build passed 😄 https://dnceng.visualstudio.com/internal/_build/results?buildId=642072&view=results

**Comment by mmitche** [comment]
Looks great from my check.

**Pull request merged**